crates/sui-sdk/src/transaction_builder.rs

```diff
@@ -133,6 +133,56 @@ impl TransactionBuilder {
         Ok(data)
     }
 
+    pub async fn pay_sui(
+        &self,
+        signer: SuiAddress,
+        input_coins: Vec<ObjectID>,
+        recipients: Vec<SuiAddress>,
+        amounts: Vec<u64>,
+        gas_budget: u64,
+    ) -> anyhow::Result<TransactionData> {
+        fp_ensure!(!input_coins.is_empty(), SuiError::EmptyInputCoins.into());
+        let gas_object = self.get_object_ref(input_coins[0]).await?;
+
+        let handles: Vec<_> = input_coins
```

Comment:
Seems like you are getting the object ref of the first object twice. I guess you could obtain the gas object ref after the handles join?